### PR TITLE
Fix/custom icons

### PR DESCRIPTION
**Pull request by AlyshaNolan**
# Context

Link to [Notion ticket](https://www.notion.so/cloudcannon/Custom-icons-d14e257f8f42459bb77ec036ce47ef7e?pvs=4) - also fixes [this ticket](https://www.notion.so/cloudcannon/Social-icons-not-working-on-safari-b2b5d59858844d3d9ba6e01589a81b63?pvs=4)

# Additional information

- Check that icons work in safari
- Check custom icon with svg and another format of image

# Testing (tester)

The reviewer should check on this branch:

- The code
- The component in Bookshop browser
- The site in CloudCannon ([link here](https://app.cloudcannon.com/42158/editor#sites/118811/collections/pages/:/edit?editor=visual&url=%2F&path=%2Fsrc%2Fpages%2Findex.html&collection=pages))

# Before merge (PR owner)

- [ ] Delete the test site in CloudCannon
- [ ] For "generic" components **only**: remove `- content_blocks` from `structures` in the YAML file (this was added for testing).

### CODE REVIEW

File: component-library/components/generic/custom-icon/custom-icon.bookshop.yml
Comment (calumocannon) [inline component-library/components/generic/custom-icon/custom-icon.bookshop.yml]:
```suggestion
  description: An icon (that you upload) with an optional border and customisable color
```

**Review by calumocannon — APPROVED** [review]
LGTM 🚀

**Comment by adon-cloudcannon** [comment]
This is fine as is... happy to approve what is here... some questions though about how this is to be used in the template and what options should be available... is this a dev component mostly? Like, when used in a component the options will be restricted to the user? Specifically:

- do the solid/outline options need to be there with this?
- is there anyway to have the user be able to choose a colour, but ALSO somehow be able to select the theme colours? Feels like this should default to choosing from the theme text colours, and then be able to change if they want?

**Review by adon-cloudcannon — COMMENTED** [review]
Actually - before approve: is this the intended behaviour for uploading custom svg?
https://talented-mongoose.cloudvent.net/

I have uploaded 2 svg:
- first is a checkmark... it was originally green... it appears to recolor inside CC, but the live site is not working as intended (still green, with a slight recolored area?)
- second is an svg with multiple colors that have all flattened into one due to the recoloring...

**Review by calumocannon — CHANGES_REQUESTED** [review]
Revert Adon's test commits before merging and fix merge conflict

File: component-library/components/generic/custom-icon/custom-icon.bookshop.yml
Comment (calumocannon) [inline component-library/components/generic/custom-icon/custom-icon.bookshop.yml]:
```suggestion
  color: "#7d7f7c"
```
The default white ish on a white background is really hard to see. Could we change this to a grey that is visible ish on both dark/light backgrounds ?

**Comment by AlyshaNolan** [comment]
Everything should be fixed now - requested re review and will remove uploads and components from index page before merging

**Review by calumocannon — APPROVED** [review]
Nice, thank you for the changes 🙌 

Approved assuming merge conflicts are resolved and test commits are reverted 🚀

**Review by adon-cloudcannon — APPROVED** [review]
Approve providing the test images are removed :)

**Pull request merged**
